include/DataFrame/Internals/DataFrame_read.tcc

```diff
@@ -1153,6 +1153,12 @@ read_csv2_(std::FILE *stream,
                     }
                 }
                 else if (col_spec.type_spec == "string")  {
+                    if (!value.empty() && value.back() == '\n') {
+                        value.pop_back();
+                    }
+                    if (!value.empty() && value.back() == '\r') {
+                        value.pop_back();
+                    }

```

Comment:
Would you mind to generalize this for `csv` and `json` formats as well? You can leave the `binary` format as is
Thanks

Reply:
Yes, sure. Will also check if I can add some test cases.

Reply:
Ok, I guess the asserts fail because of floating point precision.